### PR TITLE
Feat/connect authenticate device (T2B1)

### PR DESCRIPTION
**Pull request by szymonlesisz**
## Description

Validate `T2B1` device authenticity. implementation of https://github.com/trezor/trezor-firmware/pull/3255

I didn't find any suitable library to parse `x509` certificate and validate certificate pubKeys/signatures without adding tons of new crypto related dependencies (the missing half of the internet which we dont have here yet) so i ended up with writing our own using nodejs `crypto` polyfilled by `crypto-browserify`

It works in browser and nodejs but Im not sure if its working in react-native

## TODO:

- [x] update new protobuf messages after merge of https://github.com/trezor/trezor-firmware/pull/3255
- [x] add production rootPubKey and caPubKeys to config
- [x] serialize/prefix ECDSA P-256 pubKey to ASN format
- [x] validate current device model capabilities https://github.com/trezor/trezor-suite/issues/8952
- [x] fetch `caPubkeys` from file and pass it as parameter of `authenticateDevice` method
- [x] add unit tests for `x509` module
- [ ] add e2e tests after merge of https://github.com/trezor/trezor-firmware/pull/3255
- [x] add docs

### CODE REVIEW

File: packages/connect/src/api/firmware/verifyAuthenticityProof.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        return {
+            valid: false,
+            caPubKey,
+            error: outdated ? 'OUTDATED_CONFIG' : 'ROOT_PUBKEY_NOT_FOUND',
+        };
+    }
+
+    // 4. validate DEVICE certificate subject (Trezor features internal_model)
+    const [subject] = deviceCert.tbsCertificate.subject;
+    if (!subject.parameters || subject.algorithm !== '2.5.4.3') {
```
Comment (matejkriz) [inline packages/connect/src/api/firmware/verifyAuthenticityProof.ts:109]: would be nice to add link in comment https://www.alvestrand.no/objectid/2.5.4.3.html

Comment (szymonlesisz) [reply packages/connect/src/api/firmware/verifyAuthenticityProof.ts:109]: done

File: packages/suite/src/middlewares/suite/sentryMiddleware.ts [hunk 2]
```diff
@@ -122,6 +123,10 @@ const sentryMiddleware =
                 });
                 break;
             }
+            case deviceAuthenticityActions.result.type:
+                if (action.payload.result.valid) return;
+                setSentryContext('device-authenticity', action.payload.result);
```
Comment (matejkriz) [inline packages/suite/src/middlewares/suite/sentryMiddleware.ts:128]: TODO: this is not reporting to sentry, just adding the context for future errors...

**Comment by matejkriz** [comment]
closing in favour of https://github.com/trezor/trezor-suite-private/pull/42 where are all the commits from here.

validation of current device model capabilities mentioned as comment in prerequisite issue

e2e tests still TODO

**Pull request closed without merging**